From 921d24531541db0d6577877973279d8e0ac02619 Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Mon, 10 Feb 2025 17:20:18 +0100
Subject: [PATCH 1533/1695] mk/compile.mk: fix header dependency in .d file

Dependency file .o.d are not updated when a header file is removed from
a source file.
Add option -MP used to fix error generated when removing header files.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 627f76d2cf7ebcc82e9667de59bf8d189ac46ef6 mk/compile.mk: fix header dependency in .d file]
Change-Id: Idf494a09b3a41f6dd2e67e8f531c75bb0f5ebec4
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
(cherry picked from commit acdc32afe18f8b71915184a01e7279ffcbe0ec20)
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/436456
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Antonio Maria BORNEO <antonio.borneo@foss.st.com>
Reviewed-by: Etienne CARRIERE <etienne.carriere@foss.st.com>
Tested-by: Antonio Maria BORNEO <antonio.borneo@foss.st.com>
---
 mk/compile.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mk/compile.mk b/mk/compile.mk
index 0a2796d78..256a8c0cf 100644
--- a/mk/compile.mk
+++ b/mk/compile.mk
@@ -122,7 +122,7 @@ comp-cppflags-$2 = $$(filter-out $$(CPPFLAGS_REMOVE) $$(cppflags-remove) \
 		      $$(cppflags-lib$$(comp-lib-$2)) $$(cppflags-$2)) \
 		      -D__FILE_ID__=$$(subst -,_,$$(subst /,_,$$(subst .,_,$1)))
 
-comp-flags-$2 += -MD -MF $$(comp-dep-$2) -MT $$@
+comp-flags-$2 += -MD -MF $$(comp-dep-$2) -MP -MT $$@
 comp-flags-$2 += $$(comp-cppflags-$2)
 
 comp-cmd-$2 = $$(comp-compiler-$2) $$(comp-flags-$2) -c $$< -o $$@
-- 
2.47.2

